src/arkode/arkode_lsrkstep_impl.h

```diff
@@ -0,0 +1,140 @@
+/*---------------------------------------------------------------
+ * Programmer(s): Mustafa Aggul @ SMU
+ *---------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2024, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ *---------------------------------------------------------------
+ * Implementation header file for ARKODE's LSRK time stepper
+ * module.
+ *--------------------------------------------------------------*/
+
+#ifndef _ARKODE_LSRKSTEP_IMPL_H
+#define _ARKODE_LSRKSTEP_IMPL_H
+
+#include <arkode/arkode_lsrkstep.h>
+
+#include "arkode_impl.h"
+
+#ifdef __cplusplus /* wrapper to enable C++ usage */
+extern "C" {
+#endif
+
+/*===============================================================
+  LSRK time step module data structure
+  ===============================================================*/
+
+/*---------------------------------------------------------------
+  Types : struct ARKodeLSRKStepMemRec, ARKodeLSRKStepMem
+  ---------------------------------------------------------------
+  The type ARKodeLSRKStepMem is type pointer to struct
+  ARKodeLSRKStepMemRec.  This structure contains fields to
+  perform an explicit Runge-Kutta time step.
+  ---------------------------------------------------------------*/
+typedef struct ARKodeLSRKStepMemRec
+{
+  /* LSRK problem specification */
+  ARKRhsFn fe;
+  ARKRhsFn fi;
+  ARKDomEigFn extDomEig;
+
+  /* LSRK method storage and parameters */
+  // N_Vector  temp1; /* Temp vector storage */
+  // N_Vector  temp2; /* Temp vector storage */
+  N_Vector* Fe; /* RHS vector storage */
```

Comment:
While this is an array of vectors, one one element is ever used. Can this be replaced with an NVector or is this to support future extensions?

Reply:
I think we kept it in this way to support future extensions and also to comply with the existing structure